File: PYTHON/create_library_with_header_py/use_library_py/use_library_py/use_library.py
#  *
#  * @file use_library.py
#  *
#  * @brief A basic node to use the publisher library
#  *
#  * @author Antonio Mauro Galiano
#  * Contact: https://www.linkedin.com/in/antoniomaurogaliano/
#  *
#

import rclpy
from publisher_library_py import PublisherLibrary


def main(args=None):
    rclpy.init(args=args)
    node = PublisherLibrary()
    node.get_logger().info("My Use Publisher Library ROS2 node started")
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.get_logger().info('Shutting down use_library node.')
    finally:
        node.destroy_node()
        rclpy.shutdown()
    return 0


if __name__ == '__main__':
    main()

File: CPP/create_library_with_header/use_library/src/use_library.cpp
/**
 * @file use_library.cpp
 *
 * @brief A basic node to use the publisher library
 *
 * @author Antonio Mauro Galiano
 * Contact: https://www.linkedin.com/in/antoniomaurogaliano/
 *
 */


#include "rclcpp/rclcpp.hpp"
#include "publisher_library/publisher_library.h"


int main(int argc, char ** argv)
{
  rclcpp::init(argc,argv);
  auto node = std::make_shared<PublisherLibrary>();
  RCLCPP_INFO(node->get_logger(), "My Use Publisher Library ROS2 node started");
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
